File: cloud_testing/cloud_api_poc/src/main/java/cloud_api_poc/StopWatch.java
package cloud_api_poc;

import java.util.concurrent.TimeUnit;

public class StopWatch {

  private long startTime = 0;
  private long endTime = 0;

  public StopWatch() {}

  public StopWatch start() {
    startTime = System.currentTimeMillis();
    return this;
  }

  /** ms to minutes */
  private long minutes(long duration) {
    return TimeUnit.MILLISECONDS.toMinutes(duration);
  }

  /** ms to seconds */
  private long seconds(long duration) {
    return TimeUnit.MILLISECONDS.toSeconds(duration);
  }

  public String end() {
    endTime = System.currentTimeMillis();
    long duration = endTime - startTime;

    long minutes = minutes(duration);
    long seconds = seconds(duration) % 60;

    return String.format("%dm %ds", minutes, seconds);
  }

  public StopWatch reset() {
    startTime = 0;
    endTime = 0;
    return this;
  }
}

File: cloud_testing/kotlin_poc/src/main/kotlin/StopWatch.kt
import java.util.concurrent.TimeUnit

class StopWatch {

    private var startTime: Long = 0
    private var endTime: Long = 0

    fun start(): StopWatch {
        startTime = System.currentTimeMillis()
        return this
    }

    /** ms to minutes  */
    private fun minutes(duration: Long): Long {
        return TimeUnit.MILLISECONDS.toMinutes(duration)
    }

    /** ms to seconds  */
    private fun seconds(duration: Long): Long {
        return TimeUnit.MILLISECONDS.toSeconds(duration)
    }

    fun end(): String {
        endTime = System.currentTimeMillis()
        val duration = endTime - startTime

        val minutes = minutes(duration)
        val seconds = seconds(duration) % 60

        return String.format("%dm %ds", minutes, seconds)
    }

    fun reset(): StopWatch {
        startTime = 0
        endTime = 0
        return this
    }
}
